ruby/src/main/java/com/google/protobuf/jruby/RubyMessage.java

```diff
@@ -346,8 +346,10 @@ public IRubyObject toHash(ThreadContext context) {
             IRubyObject value = getField(context, fdef);
             if (value.respondsTo("to_h")) {
                 value = Helpers.invoke(context, value, "to_h");
+            } else if (value.respondsTo("to_a")) {
+                value = Helpers.invoke(context, value, "to_a");
             }
-            ret.fastASet(runtime.newString(fdef.getName()), value);
+            ret.fastASet(runtime.newSymbol(fdef.getName()), value);
```

Comment:
**NOTE** this is a breaking change.  It brings it inline with the initializers, and it is quite a bit faster, but it is a breaking change at that...